plugin/pkg/admission/podpreset/admission.go

```diff
@@ -130,6 +130,29 @@ func (c *podPresetPlugin) Admit(a admission.Attributes) error {
 
 		glog.V(4).Infof("PodPreset %s matches pod %s labels", pip.GetName(), pod.GetName())
 
+		//merge in policy for Containers
+		if pip.Spec.Containers != nil {
+			r, err := mergeContainers(pip, pod.Spec.Containers)
+			if err != nil {
+				// add event to pod
+				c.addEvent(pod, pip, err.Error())
+
+				return nil
+			}
+			pod.Spec.Containers = r
+		}
+
+		if pip.Spec.InitContainers != nil {
+			r, err := mergeInitContainers(pip, pod.Spec.InitContainers)
```

Comment:
I see that `mergeContainers()` and `mergeInitContainers()` are mostly identical and can be replaced by a single function: `mergeContainers(pip.GetName(), pip.Spec.Containers, pod.Spec.Containers)` and for init containers: `mergeContainers(pip.GetName(), pip.Spec.InitContainers, pod.Spec.InitContainers)`